commands/create.go

```diff
@@ -550,9 +573,16 @@ func replaceUserdataFile(userdataContent, customScriptContent []byte, newUserDat
 			return err
 		}
 
+		encodedCustomInstallScript, err := gzipEncode(modifiedCustomScriptContent)
+		if err != nil {
+			return err
+		}
+
+
 		// Add to the write_files directive
 		writeFile := map[string]string{
-			"content":     string(customScriptContent),
+			"encoding":    "gzip+b64",
+			"content":     fmt.Sprintf("!!binary | \n %s",encodedCustomInstallScript),
```

Comment:
Technically, `encodedCustomInstallScript` is not encoded because you appended code to decode it on the fly.